src/index.js

```diff
@@ -281,6 +295,12 @@ function attachSourceIfNecessary ({ nodes }, selector) {
   }
 }
 
+function assertIsDocumentOrShadowRoot (context) {
+  if (context.nodeType !== 11 && context.nodeType !== 9) {
```

Comment:
I realized duck typing this will be a cleaner approach _and_ work across windows.